apps/dashboard/app/api/auth/[...nextauth]/route.ts

```diff
@@ -1,57 +1,64 @@
+import NextAuth, { AuthOptions } from "next-auth";
 import { fetchUserData } from "@/services/graphql/queries/me-data";
-import NextAuth from "next-auth"
+import { env } from "@/env";
+import { ApolloQueryResult } from "@apollo/client";
+import { MeQuery } from "@/services/graphql/generated";
 
-const type = "oauth" as const // as ProviderType
+declare module "next-auth" {
+  interface Profile {
+    id: string;
+  }
+  interface Session {
+    sub: string | null;
+    accessToken: string;
+    userData: ApolloQueryResult<MeQuery>;
+  }
+}
 
-export const authOptions = {
-  // Configure one or more authentication providers
+const type = "oauth" as const;
+export const authOptions: AuthOptions = {
   providers: [
     {
       id: "blink",
-      clientId: process.env.CLIENT_ID,
-      clientSecret: process.env.CLIENT_SECRET,
-      wellKnown: "http://127.0.0.1:4444/.well-known/openid-configuration",
-      useSecureCookies: false,
+      clientId: env.CLIENT_ID,
+      clientSecret: env.CLIENT_SECRET,
+      wellKnown: env.HYDRA_PUBLIC,
```

Comment:
```suggestion
      wellKnown: `${env.HYDRA_PUBLIC}/.well-known/...`,
```

Reply:
my point in my prior comment is not to replace the env variable name, but also the idea to just use the `origin`, not the `pathname`. so that if there are multiple place that need to refer hydra for different endpoint, we don't need to set the env variables multiple times.